.github/workflows/e2e.maven.workflow_dispatch.main.default.slsa3.yml

```diff
@@ -0,0 +1,98 @@
+name: Maven e2e test - simple # TODO: Remove name
+on:
+  - workflow_dispatch
+# TODO: Un-comment this     
+#  schedule:
+#    - cron: "0 6 * * *"
+#  workflow_dispatch:
+#
+permissions: read-all
+
+env:
+  GH_TOKEN: ${{ github.token }}
+
+jobs:
+  bootstrap:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'schedule' || github.event_name == 'workflow_dispatch'
+    permissions:
+      contents: write
+    outputs:
+      artifact-version: ${{ steps.maven-push.outputs.artifact-version }}
+    steps:
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+      - env:
+          PACKAGE_DIR: ./e2e/maven
+        id: maven-push
+        run: ./.github/workflows/scripts/e2e-maven-push.sh
+
+  if-bootstrap-failed:
+    runs-on: ubuntu-latest
+    needs: [bootstrap]
+    if: always() && (github.event_name == 'schedule' || (github.event_name == 'workflow_dispatch' && !inputs.trigger_build)) && needs.bootstrap.result != 'success'
+    steps:
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+      - run: ./.github/workflows/scripts/e2e-report-failure.sh
+  
+  # Main workflow
+  ################################################################################
+  # Shim determines if the rest of the workflow should run.
+  # NOTE: it should only use the `if` to determine this and all downstream jobs
+  #      should depend on this job.
+  shim:
+    # NOTE: this must be kept in sync with the if-failed job.
+    if: github.event_name == 'workflow_dispatch' && inputs.trigger_build
+    runs-on: ubuntu-latest
+    steps:
+      - run: |
+          echo "event: ${GITHUB_EVENT_NAME}"
+          echo "ref: ${GITHUB_REF}"
+
+  build:
+    permissions:
+      id-token: write # For signing.
+      contents: read # For repo checkout of private repos.
+      actions: read # For getting workflow run on private repos.
+    uses: AdamKorcz/slsa-github-generator/.github/workflows/builder_maven_slsa3.yml@maven-builder-test-updates
+    with:
+      directory: ./e2e/maven
+
+  verify:
+    runs-on: ubuntu-latest
+    needs: [build, bootstrap]
+    steps:
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+      - uses: slsa-framework/slsa-github-generator/.github/actions/secure-download-folder@main
```

Comment:
here we should mimic what a user would do, using only the public download Action for the builder. If the Action does not exist, we need to create it.  See example for nodejs https://github.com/slsa-framework/slsa-github-generator/tree/main/actions/nodejs

If we don't expect users to download the attestations, then we don't need to make a public Action for it. Is this the assumption for now?

Reply:
Right. technically speaking the actions under `.github/actions` are not publicly supported for users to use and we might change their API at any time. Though GitHub will allow them to be called.

Reply:
Should be fixed by https://github.com/slsa-framework/slsa-github-generator/pull/2613